### PR TITLE
Fix reset of attributes when moving a task to a project

### PR DESCRIPTION
**Pull request by yasakova-anastasia**
### Motivation and context

- Fixed reset of attributes when moving a task to a project

Steps to reproduce the bug:

1. Create a task with a couple of labels and attributes
2. Annotate several frames in the task. Attributes should have non-default value.
3. Create a project and copy the definition of labels with attributes
4. Move the task to the project.
5. Attributes should have default values after the action

### How has this been tested?

Added REST API test

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by kirill-sizov** [comment]
I see some strange behaviour try to create task with the following label configuration:
<details>
<summary>task configuration</summary>
```
[
  {
    "name": "car",
    "id": 9,
    "color": "#2080c0",
    "type": "any",
    "attributes": [
      {
        "id": 5,
        "name": "mark",
        "input_type": "select",
        "mutable": false,
        "values": [
          "bmw",
          "audi",
          "fiat"
        ],
        "default_value": "bmw"
      }
    ]
  }
]
```
</details>

Annotate some frames and set some non-default attributes, then create project with the following label configuration
<details>
<summary>project configuration</summary>
[
  {
    "name": "car",
    "id": 10,
    "color": "#2080c0",
    "type": "any",
    "attributes": [
      {
        "id": 6,
        "name": "mark",
        "input_type": "select",
        "mutable": false,
        "values": [
          "ford",
          "honda",
          "volvo"
        ],
        "default_value": "ford"
      }
    ]
  }
]
```
</details>

Then move annotated task into the project.

It's questionable should we allow to move task in project if project has the same label with the same name attributes but with different attributes spec

**Comment by yasakova-anastasia** [comment]
> It's questionable should we allow to move task in project if project has the same label with the same name attributes but with different attributes spec

I think that the task with such attributes should not be moved. Expanded the check for this.

File: tests/python/rest_api/test_tasks.py [hunk 1]
```diff
@@ -1025,7 +1025,7 @@ def test_cannot_rename_label_to_duplicate_name(self, tasks, labels, admin_user):
         assert "All label names must be unique" in response.text
 
     def test_cannot_add_foreign_label(self, tasks, labels, admin_user):
-        task = list(tasks)[0]
+        task = [t for t in tasks if t["project_id"] is None][0]
```
Comment (kirill-sizov) [inline tests/python/rest_api/test_tasks.py:1028]: Why do we need this condition?

Comment (yasakova-anastasia) [reply tests/python/rest_api/test_tasks.py:1028]: We cannot add a label to a task that belongs to a project. This check avoids this.

**Review by kirill-sizov — APPROVED** [review]
LGTM

**Review by nmanovic — APPROVED** [review]

**Pull request merged**